1-Draft/RFCnnnn-PowerShell-Configuration.md

```diff
@@ -0,0 +1,185 @@
+Registry key | Software\Policies\PowerShellCore\OtherPowerShellApplication | Software\PowerShellCore\OtherPowerShellApplication
+
+### Priority of applying settings
+
+Because a configuration setting can be in several schemes, the setting wins according to the priority of its scheme.
+
+#### Priorities for Policy settings in descending order
+
+Scheme | Windows | Unix
+-| - | -
+GPO -> Computer Policy | HKLM\Software\Policies\PowerShellCore | /etc/powershell.config.json
+GPO -> User Policy | HKCU\Software\Policies\PowerShellCore | See `Comment A` below
+File -> Computer-Wide | %ProgramFiles%/PowerShell/powershell.config.json | /opt/Microsoft/powershell/powershell.config.json
+File -> Application-Startup | pwsh -settingsfile `somepath/powershell.config.json` | pwsh -settingsfile `somepath/powershell.config.json`
+File -> User-Wide | %APPDATA%/powershell.config.json | %XDG_CONFIG_HOME%/powershell.config.json
+File -> Application-Wide | $apphome/powershell.config.json | $apphome/powershell.config.json
+
+Defaults:
+
+`%APPDATA%` - `C:\Users\useraccount\AppData\Roaming`
+
+`%XDG_CONFIG_HOME%` - `HOME/.config`
+
+#### Parameter `-settingsfile`
+
+With `-settingsfile` parameter users can assign custom settings from the config file and overwrite user-wide and application-wide settings. Only users with elevated rights can overwrite computer-wide and user policy settings.
+
+#### Priorities for Regular settings in descending order
+
+Scheme | Windows | Unix
+-| - | -
+File -> Application-Startup | pwsh -settingsfile `somepath/powershell.config.json` | pwsh -settingsfile `somepath/powershell.config.json`
+File -> Application-Wide | $apphome\powershell.config.json | $apphome/powershell.config.json
+File -> User-Wide | %APPDATA%\powershell.config.json | ~/powershell.config.json
+File -> Computer-Wide | %ProgramFiles%\PowerShell\powershell.config.json | /opt/Microsoft/powershell/powershell.config.json
+GPO -> User Config | HKCU\Software\PowerShellCore | See `Comment A` below
+GPO -> Computer Config | HKLM\Software\PowerShellCore | /etc/powershell.config.json
+
+### Configuration settings
+
+A set of configuration settings in GPO scheme and file scheme for policy settings and regular settings is the same. This allows to discover and configure settings in the simplest and fastest way.
+
+#### Registry keys and settings
+
+| Key | SubKey | Option | Type
+| -| - | - | -
+Software\Policies\PowerShellCore | - | -
+Software\PowerShellCore | - | -
+| | | ExecutionPolicy | String
+| | | PipelineMaxStackSizeMB | DWORD
+| | ConsoleSessionConfiguration | EnableConsoleSessionConfiguration | DWORD
+| | ConsoleSessionConfiguration | ConsoleSessionConfigurationName | String
+| | ModuleLogging | EnableModuleLogging | DWORD
+| | ModuleLogging | ModuleNames | String
+| | ProtectedEventLogging | EncryptionCertificate | DWORD
+| | ScriptBlockLogging | EnableScriptBlockInvocationLogging | DWORD
+| | ScriptBlockLogging | EnableScriptBlockLogging | DWORD
```

Comment:
Can `ScriptBlockLogging\*` and `ProtectedEventLogging\*` also be configured via policy?

Reply:
These setting plus execution policy are Security related.

Reply:
ModuleLogging also..

Transcription would not be set system wide, it would need to be per user or  it's better to use scriptblock logging. IMHO.

cc @PaulHigin